app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -0,0 +1,361 @@
+package com.linkedin.drelephant.tuning.engine;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import models.AppResult;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+
+import com.linkedin.drelephant.spark.heuristics.ConfigurationHeuristic;
+import com.linkedin.drelephant.spark.heuristics.DriverHeuristic;
+import com.linkedin.drelephant.spark.heuristics.ExecutorGcHeuristic;
+import com.linkedin.drelephant.spark.heuristics.JvmUsedMemoryHeuristic;
+import com.linkedin.drelephant.spark.heuristics.UnifiedMemoryHeuristic;
+import com.linkedin.drelephant.util.MemoryFormatUtils;
+
+
+/**
+ * This class recommends spark parameters based on previous run heuristics. For spark it expects one
+ */
+public class SparkHBTParamRecommender {
+  private final Logger logger = Logger.getLogger(SparkHBTParamRecommender.class);
+
+  AppResult appResult;
+
+  // TODos Move these to configuration
+  public static final int MAX_EXECUTOR_CORE = 3;
+  public static final long MAX_EXECUTOR_MEMORY = 10 * FileUtils.ONE_GB;
+  public static final long MIN_EXECUTOR_MEMORY = 900 * FileUtils.ONE_MB;
```

Comment:
Default min memoryOverhead is 384MB -- if the goal is 1GB container, then min would be 640MB. I'm using this in the version I have. However, the cluster default executor memory is 2GB, and I am not sure if 640MB might be too low.